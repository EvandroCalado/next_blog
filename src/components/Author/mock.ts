export default {
  title: 'Evandro Calado',
  resume: 'Desenvolvedor Frontend',
  image: 'https://avatars.githubusercontent.com/u/110628201?v=4',
  alt: 'Evandro Calado',
};
