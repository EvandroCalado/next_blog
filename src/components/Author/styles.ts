import styled, { css } from 'styled-components';

export const Wrapper = styled.div`
${({ theme }) => css`
  width: 100%;
  display: flex;
  align-items: center;
  justify-content: center;
  gap: 3rem;
  margin: ${theme.spacings.xxlarge} 0;
`}
`;

export const Image = styled.img`
  ${({ theme }) => css`
    width: 15rem;
    border: 3px solid ${theme.colors.secondary};
    border-radius: 50%;
  `}
`;

export const Container = styled.div`
`;

export const Title = styled.h2`
${({ theme }) => css`
font-size: ${theme.font.sizes.xlarge};
`}
`;

export const Paragraph = styled.p`
`;
