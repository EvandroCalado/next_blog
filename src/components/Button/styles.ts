import styled, { css } from 'styled-components';
import { ButtonProps } from './';

export const Button = styled.button<Pick<ButtonProps, 'color'>>`
  ${({ theme, color }) => css`
    background-color: ${theme.colors[color]};
    color: ${theme.colors.white};
    font-size: ${theme.font.sizes.medium};
    border: none;
    padding: ${theme.spacings.xsmall} ${theme.spacings.medium};
    margin: ${theme.spacings.large} 0;
    border-radius: 0.5rem;
    display: flex;
    align-items: center;
    justify-content: center;
    transition: all 300ms ease-in-out;
    cursor: pointer;

    &:hover {
      filter: brightness(80%);
    }

    &:disabled {
      background-color: ${theme.colors.mediumGray};
      color: ${theme.colors.darkGray};
      cursor: not-allowed;

      &:hover {
        filter: none;
      }
    }

    > svg {
      width: 3rem;
      margin-left: 1rem;
    }
  `}
`;
