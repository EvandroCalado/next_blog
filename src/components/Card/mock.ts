export default {
  slug: 'slug-test',
  title: 'Title',
  cover:
    'https://images.unsplash.com/photo-1593720219276-0b1eacd0aef4?ixlib=rb-4.0.3&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=640&q=80',
  excerpt:
    'JavaScript é uma linguagem de programação popular e amplamente utilizada para criar e desenvolver aplicativos web dinâmicos e interativos. Ela é interpretada pelo navegador do usuário e é executada em tempo real, permitindo que os desenvolvedores adicionem funcionalidades avançadas aos sites, como animações, interatividade do usuário, validação de formulários, solicitações assíncronas ao servidor e muito mais.',
};
