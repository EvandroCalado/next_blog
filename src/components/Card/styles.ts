import styled, { css } from 'styled-components';

export const Wrapper = styled.div`
    a {
      transition: all 300ms ease-in-out;
    }

    a:hover {
      opacity: 1;
      filter: brightness(170%);
    }
`;

export const Cover = styled.div`
  ${({ theme }) => css`
    img {
      width: 100%;
      display: block;
      aspect-ratio: 16/9;
      border-radius: .5rem;
      margin-bottom: ${theme.spacings.small};

    }
  `}
`;

export const Heading = styled.h2`
  ${({ theme }) => css`
    font-size: ${theme.font.sizes.smallMedium};
    color: ${theme.colors.darkGray};
    margin-bottom: ${theme.spacings.xsmall};
  `}
`;

export const Paragraph = styled.p`
  ${({ theme }) => css`
    font-size: ${theme.font.sizes.small};
    margin-bottom: ${theme.spacings.small};
    color: ${theme.colors.primary};
  `}
`;
