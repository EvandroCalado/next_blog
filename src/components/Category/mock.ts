export default {
  categories: [
    {
      image:
        'https://res.cloudinary.com/dnhsdenji/image/upload/v1682864831/javascript_programming_language_icon_ce8f9d2d57.png',
      title: 'Javascript',
    },
    {
      image:
        'https://res.cloudinary.com/dnhsdenji/image/upload/v1682864831/javascript_programming_language_icon_ce8f9d2d57.png',
      title: 'Javascript',
    },
  ],
  category: 'Javascript',
};
