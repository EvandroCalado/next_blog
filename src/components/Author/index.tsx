import { PostStrapi } from '../../typing/posts';
import * as Styled from './styles';

export type AuthorProps = {
  post: PostStrapi;
};

const Author = ({ post }: AuthorProps) => {
  const image =
    post.attributes.author.data.attributes.image.data.attributes.url;
  const alt = post.attributes.author.data.attributes.image.data.attributes.name;
  const title = post.attributes.author.data.attributes.name;
  const resume = post.attributes.author.data.attributes.resume;

  return (
    <Styled.Wrapper>
      <Styled.Image src={image} alt={alt} />
      <Styled.Container>
        <Styled.Title>{title}</Styled.Title>
        <Styled.Paragraph>{resume}</Styled.Paragraph>
      </Styled.Container>
    </Styled.Wrapper>
  );
};

export default Author;
