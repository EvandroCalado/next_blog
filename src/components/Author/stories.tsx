import Author, { AuthorProps } from './';
import { Meta, StoryFn } from '@storybook/react';

import mock from './mock';

export default {
  title: 'Author',
  component: Author,
  args: mock,
} as Meta;

export const Template: StoryFn<AuthorProps> = (args) => (
  <div>
    <Author {...args} />
  </div>
);
