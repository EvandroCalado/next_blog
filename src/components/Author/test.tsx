import Author from './';
import { screen } from '@testing-library/react';
import { renderTheme } from '../../styles/renderTheme';

import mock from './mock';

describe('<Author />', () => {
  it('should render image', () => {
    renderTheme(<Author {...mock} />);
    expect(screen.getByRole('img', { name: mock.title })).toBeInTheDocument();
  });

  it('should render title', () => {
    renderTheme(<Author {...mock} />);
    expect(screen.getByText(mock.title)).toBeInTheDocument();
  });

  it('should render paragraph', () => {
    renderTheme(<Author {...mock} />);
    expect(screen.getByText('Desenvolvedor Frontend')).toBeInTheDocument();
  });

  it('should match snapshot', () => {
    const { container } = renderTheme(<Author {...mock} />);
    expect(container).toMatchSnapshot();
  });
});
