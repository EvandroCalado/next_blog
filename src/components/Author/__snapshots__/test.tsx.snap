// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Author /> should match snapshot 1`] = `
.c0 {
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  gap: 3rem;
  margin: 4.8rem 0;
}

.c1 {
  width: 15rem;
  border: 3px solid #f24b4b;
  border-radius: 50%;
}

.c2 {
  font-size: 4.0rem;
}

<div>
  <div
    class="c0"
  >
    <img
      alt="Evandro Calado"
      class="c1"
      src="https://avatars.githubusercontent.com/u/110628201?v=4"
    />
    <div
      class=""
    >
      <h2
        class="c2"
      >
        Evandro Calado
      </h2>
      <p
        class=""
      >
        Desenvolvedor Frontend
      </p>
    </div>
  </div>
</div>
`;
